openmdao/approximation_schemes/finite_difference.py

```diff
@@ -117,6 +121,23 @@ def add_approximation(self, abs_key, system, kwargs, vector=None):
                                  "one of {}".format(system.msginfo, form,
                                                     list(DEFAULT_ORDER.keys())))
 
+            step_calc = options['step_calc']
+            step_calcs = ['abs', 'rel', 'rel_legacy', 'rel_avg', 'rel_element']
+            if step_calc not in step_calcs:
+                raise ValueError(f"{system.msginfo}: '{step_calc}' is not a valid setting for "
+                                 f"step_calc; must be one of {step_calcs}.")
+
+            elif options['directional'] and step_calc == 'rel_element':
+                raise ValueError(f"{system.msginfo}: Option 'directional' is not supported when "
+                                 "'step_calc' is set to 'rel_element.'")
+
+            elif step_calc == 'rel':
+                warn_deprecation("When using 'rel' as the step_calc, the fd stepsize is currently "
+                                 "scaled by the norm of the vector variable. This is not ideal for"
+                                 " larger vectors, and this behavior is being changed in "
+                                 "OpenMDAO 3.12.0. To preserve the older way of doing this "
+                                 "calculation, set step_calc to 'rel_legacy'.")
```

Comment:
It seems like we could make use of an `OptionsDictionary` to handle some of these checks (validity & deprecation)... is there a reason we wouldn't want to do that?

Reply:
I think the main reason is that the default values branch and some options become valid/invalid depending on whether you choose 'cs' or 'fd'.

Reply:
Though, if you mean specifically on the `finite_difference` object, we could do it.  I reckon this class was crafted before we started using the OptionsDictionary for call arguments.